### ISSUE TITLE
SuppressionCommentFilterTest should fail when locale is changed

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#5244]
SuppressionCommentFilterTest has [hardcoded violation messages in English](https://github.com/checkstyle/checkstyle/blob/2aca36156fa461f2d54f1386743a0146bdf8b4bd/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilterTest.java#L53-L71). When [TravisCI executes "localized" tests](https://github.com/checkstyle/checkstyle/blob/062aa888e9e789676ea03fdc926d596ce5d85224/.travis.yml#L90-L94), the tests in SuppressionCommentFilterTest should fail, because the checks which are used in Checker configuration should throw violations in specific locale, not in English.

The issue may be connected to https://github.com/checkstyle/checkstyle/issues/3377 and https://github.com/checkstyle/checkstyle/issues/3110.

I noticed the problem when I implemented tests for SuppressWithPlainTextCommentFilter in the scope of #4841. I hardcoded violation messages in UTs in English as it was done in SuppressionCommentFilterTest, but Travis CI failed [the build](https://travis-ci.org/checkstyle/checkstyle/jobs/293374566).

[Build log.](https://github.com/checkstyle/checkstyle/files/1424120/log.txt)

**Comment by romani** [comment]
this due [SuppressionCommentFilterTest#227](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilterTest.java#L227)
```
        final Checker checker = new Checker();
        final Locale locale = Locale.ROOT;
        checker.setLocaleCountry(locale.getCountry());
        checker.setLocaleLanguage(locale.getLanguage());
```

it is better to find a way make such test localized too as all others.

**Comment by romani** [comment]
@dyominov , can you help us with this issue ?

**Comment by dyominov** [comment]
I am on it

**Comment by dyominov** [comment]
In my opinion if change 
`        final Locale locale = Locale.ROOT;
`
to 
`        final Locale locale = Locale.getDefault();
`
this can solve problem.
But i'm not sure.

**Comment by dyominov** [comment]
I have 2 ways to solve this issue

- 1 : change 
`final Locale locale = Locale.ROOT;`
to 
` final Locale locale = Locale.getDefault();`

but if i change my local to "fr" i take this  

> Expected 13:17: Name 'I' must match pattern '^[a-z][a-zA-Z0-9]*$'.

> Actual 13:17: Le nom 'I' n'est pas conforme à l'expression '^[a-z][a-zA-Z0-9]*$'.

and another local too. Need create [allMessage](https://github.com/checkstyle/checkstyle/blob/2aca36156fa461f2d54f1386743a0146bdf8b4bd/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilterTest.java#L53-L71) for all language and add to test.

- 2: change 

`final Locale locale = Locale.ROOT;`
to 
` final Locale locale = Locale.ENGLISH;`

and it pass all test.

Which way me need to solve this issue or need another way?

**Comment by romani** [comment]
@dyominov ,

we need to remove set of locale at all and generate expected string base on current locale, as all other UTs do.

**Comment by rnveach** [comment]
@romani I am closing this issue as `SuppressionCommentFilterTest` was done in the scope of https://github.com/checkstyle/checkstyle/issues/5157 (specifically PR https://github.com/checkstyle/checkstyle/pull/5277)  because the tests were creating custom checkers and we should only be using the ones provided by `AbstractModuleTestSupport`.
